app/controllers/join_team_requests_controller.rb

```diff
@@ -93,4 +102,31 @@ def respond_after(request)
       format.xml { render xml: request }
     end
   end
+
+  def check_team_before_accept
+    @inviter_userid=(params[:inviter_user_id]).to_i
+    @assignment_id=(params[:invited_assignment_id]).to_i
+    # check if the inviter's team is still existing, and have available slot to add the invitee
+    inviter_assignment_team = AssignmentTeam.team(AssignmentParticipant.find_by(user_id: @inviter_userid, parent_id:@assignment_id))
+    if inviter_assignment_team.nil?
+      flash[:error] = 'The team that invited you does not exist anymore.'
+      redirect_to view_student_teams_path student_id: params[:student_id]
+    elsif inviter_assignment_team.full?
+      flash[:error] = 'The team that invited you is full now.'
+      redirect_to view_student_teams_path student_id: params[:student_id]
+    else
+      invitation_accept
+    end
+  end
+
+  def invitation_accept
```

Comment:
please explain this method in a comment. what's the different with "accept" method above